README.md

```diff
@@ -1,2 +1,118 @@
-# juttle-engine
-juttle execution engine
+# Juttle Engine
+
+[![Build Status](https://travis-ci.org/juttle/juttle-engine.svg?branch=master)](https://travis-ci.org/juttle/juttle-engine)
+
+Juttle Engine is an integrated environment for developing and executing
+[juttle](https://github.com/juttle/juttle) programs and visualizations. It lets
+you run Juttle programs stored on the file system and present the results in a
+browser for experimentation, development, deployment, and debugging of juttle
+programs.
+
+When using Juttle to power data visualizations, they cannot be rendered in the
+Juttle CLI. Juttle Engine bridges this gap by including a browser based
+execution application with a javascript charting library that receives streaming
+data from Juttle programs running server-side via a websocket. It also includes
+user-interface logic to implement Juttle input controls.
+
+![](docs/screenshots/stock_prices_example.png)
+
+If you simply want to try running some basic juttle programs that output directly to the terminal, you can use the ``juttle`` command line program included in the Juttle respository.
+
+Under the covers, Juttle Engine integrates
+[juttle-service](https://github.com/juttle/juttle-service), a node.js API server
+that enables execution of Juttle programs using a REST API with the ability to
+serve web application bundles. It comes with the
+[juttle-viewer](https://github.com/juttle/juttle-viewer) application that
+provides a simple in browser experience for loading and executing juttle
+programs in the juttle-engine and rendering the results using the
+[juttle-viz](https://github.com/juttle/juttle-viz) visualization library. In the
+future, Juttle Engine may be extended to support other application bundles.
+
+## Getting Started
+
+### Installation
+
+Make sure you have [node](http://nodejs.org) (with [npm](http://npmjs.org)) installed.
+
+Use npm to install Juttle Engine
+```
+$ npm install -g outrigger
+```
+
+This will install the Juttle Engine daemon and client binaries, the [http://github.com/juttle/juttle](juttle) interpreter, and the current set of supported juttle adapters.
+
+We've tested with nodejs 4.2.3 and npm 2.14.17. Other combinations of nodejs and npm likely work, but we haven't tested all combinations.
+
+### Running your first juttle program
+
+Start the daemon by running juttle-engine:
+```
+$ juttle-engine -d -o juttle-engine.log
+```
+This will daemonize juttle-engine and output logs to juttle-engine.log. You can now run juttle programs against the juttle-engine daemon via the juttle-engine-client:
+```
+$ juttle-engine-client browser --path my_juttle_file.juttle
+```
```

Comment:
It may help to link to an actual juttle program you could run, even a hello world example, so a newbie user is not stuck wondering "what do i put in my_juttle_file.juttle"